feat: extend kubernetes perf tuning section (#1444)
kumahq/kuma-website: app/_src/production/upgrades-tuning/fine-tuning.md

{% endwarning %}

{% if_version lte:2.1.x %}

## Kubernetes

### Kubernetes outbounds in central place

Configure `KUMA_EXPERIMENTAL_KUBE_OUTBOUNDS_AS_VIPS` to `true` to store the list of outbounds in ConfigMap that is used for VIPs of {{site.mesh_product_name}} DNS.
This way we don't repeat this information across all `Dataplane` objects which may improve a performance with a large number of data plane proxies.

You can enable this only after all instances of the control plane are updated to 1.6.0 or later.
This option will be the default behaviour in the next versions of {{site.mesh_product_name}}.
{% endif_version %}

{% if_version gte:2.3.x %}

### Kubernetes client

> Kubernetes client uses client level throttling to not overwhelm kube-api server. In larger deployments, bigger than 2k services in a single kubernetes cluster, number

Reviewer: ```suggestion
Kubernetes client uses client level throttling to not overwhelm kube-api server. In larger deployments, bigger than 2000 services in a single kubernetes cluster, number
```

of resources updates can hit this throttling. In most cases it's safe to increase this limit as kube-api have it's own throttling mechanism. To change client
throttling configuration you need to update config.

```yaml
runtime:
  kubernetes:
    clientConfig:
      qps: ... # Qps defines maximum requests kubernetes client is allowed to make per second.
      burstQps: ... # BurstQps defines maximum burst requests kubernetes client is allowed to make per second
```

### Kubernetes controller manager

> {{site.mesh_product_name}} is modifying some kubernetes resources. Kubernetes call the process of modification reconciliation. Every resource has its own working queue, and control plane adds reconciliation tasks to that queue. In larger deployments, bigger than 2k services in a single kubernetes cluster, size of the work queue for pod reconciliation

Reviewer: ```suggestion
{{site.mesh_product_name}} is modifying some Kubernetes resources. Kubernetes calls the process of modification reconciliation. Every resource has its own working queue, and control plane adds reconciliation tasks to that queue. In larger deployments, bigger than 2000 services in a single Kubernetes cluster, size of the work queue for pod reconciliation
```

> can grow and slow down pods updates. In this situation you can change the number of concurrent pod roconciliation tasks, by changing configuration:

Reviewer: ```suggestion
can grow and slow down pods updates. In this situation you can change the number of concurrent pod reconciliation tasks, by changing configuration:
```
